File: Lab_2/task_3/3.go
package main

import (
	"fmt"
)

// Функция для переворачивания числа и удаления ведущих нулей
func reverseNumber(num int) int {
	reversed := 0
	for num > 0 {
		reversed = reversed*10 + num%10
		num /= 10
	}
	return reversed
}

func main() {
	var n int
	fmt.Print("Введите количество чисел: ")
	_, err := fmt.Scan(&n)
	if err != nil || n <= 0 {
		fmt.Println("Ошибка ввода. Введите корректное количество чисел.")
		return
	}

	fmt.Printf("Введите %d чисел: ", n)
	for i := 0; i < n; i++ {
		var num int
		_, err := fmt.Scan(&num)
		if err != nil {
			fmt.Println("Ошибка ввода:", err)
			return
		}
		fmt.Print(reverseNumber(num), " ")
	}

	fmt.Println()
}

File: Lab_2/task_3/3.java
import java.util.Scanner;

import java.io.PrintStream;  import static java.nio.charset.StandardCharsets.UTF_8;

public class Main {

    // Функция для переворачивания числа и удаления ведущих нулей
    public static int reverseNumber(int num) {
        int reversed = 0;
        while (num > 0) {
            reversed = reversed * 10 + num % 10;
            num /= 10;
        }
        return reversed;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        PrintStream out = new PrintStream(System.out, true, UTF_8);

        out.print("Введите количество чисел: ");
        int n = scanner.nextInt();

        if (n <= 0) {
            out.println("Некорректный ввод.");
            return;
        }

        out.print("Введите " + n + " чисел: ");
        for (int i = 0; i < n; i++) {
            int num = scanner.nextInt();
            out.print(reverseNumber(num) + " ");
        }

        System.out.println();
        scanner.close();
    }
}
